pkg/kn/commands/service/service_describe.go

```diff
@@ -41,23 +122,515 @@ func NewServiceDescribeCommand(p *commands.KnParams) *cobra.Command {
+	dw.WriteColsLn(printers.LEVEL_0, l("Conditions"))
+	maxLen := getMaxTypeLen(service.Status.Conditions)
+	formatHeader := "%-2s %-" + strconv.Itoa(maxLen) + "s %6s %-s\n"
+	formatRow := "%-2s %-" + strconv.Itoa(maxLen) + "s %6s %-s\n"
+	dw.Write(printers.LEVEL_1, formatHeader, "OK", "TYPE", "AGE", "REASON")
+	for _, condition := range service.Status.Conditions {
+		ok := formatStatus(condition.Status)
+		reason := condition.Reason
+		if printDetails && reason != "" {
+			reason = fmt.Sprintf("%s (%s)", reason, condition.Message)
+		}
+		dw.Write(printers.LEVEL_1, formatRow, ok, formatConditionType(condition), age(condition.LastTransitionTime.Inner.Time), reason)
+	}
+}
+
+func writeConcurrencyOptions(dw printers.PrefixWriter, desc *revisionDesc) {
+	dw.WriteColsLn(printers.LEVEL_1, "", l("Concurrency"))
+	if desc.concurrencyLimit != nil {
+		dw.WriteColsLn(printers.LEVEL_2, "", "", l("Limit"), strconv.FormatInt(*desc.concurrencyLimit, 10))
+	}
+	if desc.concurrencyTarget != nil {
+		dw.WriteColsLn(printers.LEVEL_2, "", "", l("Target"), strconv.Itoa(*desc.concurrencyTarget))
+	}
+}
+
+// ======================================================================================
+// Helper functions
+
+// Format label (extracted so that color could be added more easily to all labels)
+func l(label string) string {
+	return label + ":"
+}
+
+// Format scale in the format "min ... max" with max = ∞ if not set
+func formatScale(minScale *int, maxScale *int) string {
+	ret := "0"
+	if minScale != nil {
+		ret = strconv.Itoa(*minScale)
+	}
+
+	ret += " ... "
+
+	if maxScale != nil {
+		ret += strconv.Itoa(*maxScale)
+	} else {
+		ret += "∞"
+	}
+	return ret
+}
+
+// Format the revision name along with its generation. Use colors if enabled.
+func getRevisionNameWithGenerationAndAge(desc *revisionDesc) string {
+	return desc.name + " " +
+		"[" + strconv.Itoa(desc.configurationGeneration) + "]" +
+		" " +
+		"(" + age(desc.creationTimestamp) + ")"
+}
+
+// Used for conditions table to do own formatting for the table,
+// as the tabbed writer doesn't work nicely with colors
+func getMaxTypeLen(conditions v1beta1.Conditions) int {
+	max := 0
+	for _, condition := range conditions {
+		if len(condition.Type) > max {
+			max = len(condition.Type)
+		}
+	}
+	return max
+}
+
+// Color the type of the conditions
+func formatConditionType(condition apis.Condition) string {
+	return string(condition.Type)
+}
+
+// Status in ASCII format
+func formatStatus(status corev1.ConditionStatus) string {
+	switch status {
+	case v1.ConditionTrue:
+		return "++"
+	case v1.ConditionFalse:
+		return "--"
+	default:
+		return ""
+	}
+}
+
+// Return either image name with tag or together with its resolved digest
+func getImageDesc(desc *revisionDesc) string {
+	image := desc.image
+	if printDetails && desc.imageDigest != "" {
+		digest := "(" + shortenDigest(desc.imageDigest) + ")"
+		return fmt.Sprintf("%s %s", image, digest)
```

Comment:
I prefer to put things into one place here:

```suggestion
		return fmt.Sprintf("%s (%s)", image, shortenDigest(desc.imageDigest))
```